### PR TITLE
feat(builder): bump scaleway to new sdk

### PR DESCRIPTION
**Pull request by Sh4d1**
Signed-off-by: Patrik Cyvoct <patrik@ptrk.io>

Bump the Scaleway Builder to use the new (and not deprecated) Go SDK.

### CODE REVIEW

File: builder/scaleway/config.go [hunk 1]
```diff
@@ -98,10 +125,43 @@ func (c *Config) Prepare(raws ...interface{}) ([]string, error) {
...
+		warnings = append(warnings, "token is deprecated in favor of secret_key")
+		c.SecretKey = c.Token
+	}
+
+	if c.Region != "" {
+		warnings = append(warnings, "region is deprecated in favor of zone")
+		c.Zone = c.Region
+	}
+
+	if c.AccessKey == "" {
```
Comment (kindermoumoute) [inline builder/scaleway/config.go:146]: would it make sense to use `scw.LoadEnvProfile()` for all env variables?

File: builder/scaleway/config.go [hunk 2]
```diff
@@ -160,9 +225,9 @@ func (c *Config) Prepare(raws ...interface{}) ([]string, error) {
 	}
 
 	if errs != nil && len(errs.Errors) > 0 {
-		return nil, errs
+		return warnings, errs
```
Comment (kindermoumoute) [inline builder/scaleway/config.go:228]: why do you return warnings here?

Comment (Sh4d1) [reply builder/scaleway/config.go:228]: Returning warnings even on error seems legit no? 😄

File: builder/scaleway/config.go (hunk 1 above)
Comment (Sh4d1) [reply builder/scaleway/config.go:146]: Hum yep we could add the `Profile` to the config 🤔

**Comment by SwampDragons** [comment]
Once the scaleway team gives a final sign off, I'll merge :)

File: builder/scaleway/config.go (hunk 1 above)
Comment (Sh4d1) [reply builder/scaleway/config.go:146]: It may complicate a bit tis part. I think keeping in the way it is is the easier approach

**Comment by Sh4d1** [comment]
@kindermoumoute @jerome-quere good for you?

**Review by kindermoumoute — APPROVED** [review]
LGTM

**Review by sylviamoss — APPROVED** [review]
Nice!! Thanks scaleway team for reviewing the PR 👍

**Pull request merged**